prow/pubsub/subscriber/subscriber.go

```diff
@@ -150,13 +150,13 @@ func (m *pubSubMessage) nack() {
 
 // jobHandler handles job type specific logic
 type jobHandler interface {
-	getProwJobSpec(cfg prowCfgClient, pc *config.InRepoConfigCacheHandler, pe ProwJobEvent) (*v1.ProwJobSpec, map[string]string, map[string]string, error)
+	getProwJobSpec(cfg prowCfgClient, pc *config.InRepoConfigCacheHandler, pe ProwJobEvent) (jobSpec *v1.ProwJobSpec, labels map[string]string, annotations map[string]string, err error)
```

Comment:
Nit: I was thinking more along the lines of https://yourbasic.org/golang/type-alias/. This is still better than the existing situation though, so +1.

Reply:
ack. The type alias change would be more involved imo, we can consider that in the future